### DISCUSSION TITLE
Use optimized crypto libs

### DISCUSSION THREAD

**Discussion by petersasi** [petersasi/deluge2-win-build#20] [Ideas]
From intel, called ISA-L, suggestion is here: https://github.com/arvidn/libtorrent/issues/6112
I am not sure if there already is an OpenSSL version patched for this, but since @mhertz has already kindly provided the script for bulding OpenSSL ourselves (that I plan ti incorporate) we might just as well build it with the several times faster crypto :)

**Comment by mhertz** [comment]
Thanks mate, and that sounds intriguing for sure :) The OP in that thread states significant refactoring probably would be needed to support this, which I guessed where libtorrent, but don't know if openssl or whatever, but if in libtorrent, then I'm guessing would happen in v2 only, like the wolfssl option, but would hope I'm wrong as sounds awesome indeed.

**Reply by petersasi** [reply]
I agree if it's direct LT support that will probably be slow and painful.

I was hoping OpenSSL, Intel or some openSSL clone, which I read about here https://daniel.haxx.se/blog/2021/04/02/where-is-http-3-right-now/ quicTLS from Microsoft and Akamai or BoringSSL from Google may have already created an integration for their own performance advantage.

I do understand that the primary purpose of these OpenSSL clones is the HTTP3 support.
